ejb3/src/main/java/org/jboss/as/ejb3/component/messagedriven/MessageDrivenComponentDescription.java

```diff
@@ -159,8 +159,12 @@ public void configure(final DeploymentPhaseContext context, final ComponentDescr
                     final EEApplicationClasses applicationClasses = context.getDeploymentUnit().getAttachment(Attachments.EE_APPLICATION_CLASSES_DESCRIPTION);
                     InterceptorClassDescription interceptorConfig = ComponentDescription.mergeInterceptorConfig(configuration.getComponentClass(), applicationClasses.getClassByName(description.getComponentClassName()), description, MetadataCompleteMarker.isMetadataComplete(context.getDeploymentUnit()));
 
-                    configuration.addPostConstructInterceptor(new LifecycleCMTTxInterceptor.Factory(interceptorConfig.getPostConstruct(), true), InterceptorOrder.ComponentPostConstruct.TRANSACTION_INTERCEPTOR);
-                    configuration.addPreDestroyInterceptor(new LifecycleCMTTxInterceptor.Factory(interceptorConfig.getPreDestroy(), true), InterceptorOrder.ComponentPreDestroy.TRANSACTION_INTERCEPTOR);
+                    if (interceptorConfig.getPostConstruct() != null) {
+                        configuration.addPostConstructInterceptor(new LifecycleCMTTxInterceptor.Factory(interceptorConfig.getPostConstruct(), true), InterceptorOrder.ComponentPostConstruct.TRANSACTION_INTERCEPTOR);
+                    }
+                    if (interceptorConfig.getPreDestroy() != null) {
+                        configuration.addPreDestroyInterceptor(new LifecycleCMTTxInterceptor.Factory(interceptorConfig.getPreDestroy(), true), InterceptorOrder.ComponentPreDestroy.TRANSACTION_INTERCEPTOR);
+                    }
```

Comment:
If *"must be specified for the method(s) on the bean class, rather than for a superclass or PostConstruct or PreDestroy interceptor method"* means it cannot be defined on the bean class level then the current code is correct for Singletons. I am not so familiar with reading those specs, but it seems the bean class level is missing there.

As noted below, in case of tx attribute definition on the bean class, lifecycle methods run in an unspecified tx context. Is this correct? It applies to both Singletons and Statefuls.